src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java

```diff
@@ -77,52 +85,62 @@ protected void onStreamClosed() {
-
-                return new PutComponentMetricResponse();
-            });
+        public PutComponentMetricResponse handleRequest(
+                PutComponentMetricRequest componentMetricRequest) {
+            return translateExceptions(
+                    () -> {
+                        logger.atDebug()
+                                .kv(SERVICE_NAME, serviceName)
+                                .log(
+                                        "Received putComponentMetricRequest from component "
+                                                + serviceName);
+
+                        // Authorize service name for given operation
+                        String opName = this.getOperationModelContext().getOperationName();
+                        try {
+                            doServiceAuthorization(opName, serviceName);
+                        } catch (AuthorizationException e) {
+                            logger.atError()
+                                    .kv(SERVICE_NAME, serviceName)
+                                    .log("{} is not authorized to perform operation", serviceName);
+                            throw new UnauthorizedError(e.getMessage());
+                        }
+
+                        // validate - metric name length, value is non negative etc etc
+                        List<software.amazon.awssdk.aws.greengrass.model.Metric> metricList =
+                                componentMetricRequest.getMetrics();
+                        try {
+                            validateComponentMetricRequest(opName, serviceName, metricList);
+                        } catch (IllegalArgumentException e) {
+                            logger.atError()
+                                    .kv(SERVICE_NAME, serviceName)
+                                    .log("invalid component metric request from {}", serviceName);
+                            throw new InvalidArgumentsError(e.getMessage());
+                        } catch (AuthorizationException e) {
+                            logger.atError()
+                                    .kv(SERVICE_NAME, serviceName)
+                                    .log("{} is not authorized to perform operation", serviceName);
+                            throw new UnauthorizedError(e.getMessage());
+                        }
+
+                        // Perform translations on metrics List
+                        try {
+                            final String metricNamespace = serviceName;
+                            translateAndEmit(metricList, metricNamespace);
+                        } catch (IllegalArgumentException e) {
+                            logger.atError()
+                                    .kv(SERVICE_NAME, serviceName)
+                                    .log("invalid component metric request from {}", serviceName);
+                            throw new InvalidArgumentsError(e.getMessage());
+                        } catch (Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)